### PR TITLE
Add tests for Package functions

### PR DESCRIPTION
**Pull request by erikscheurer**
This PR adds tests for the most of the package functions. The tests are heavily inspired by the [python bindings](https://github.com/precice/python-bindings/tree/develop/test).

The tests use the original Julia bindings, but make Julia use a dummy API instead of the real preCICE API.

### CODE REVIEW

**Comment by erikscheurer** [comment]
At the moment, the fake Solverdummy interface has to be specified manually for the test. This is done by running in the test directory:
```
$ make
$ julia
julia> ]
pkg> dev ../
julia> push(Libc.Libdl.DL_LOAD_PATH, "<path-to-test-directory>")
julia> ]
pkg> test PreCICE
```
Then the new testset should run, and pass except for one test. At the moment, after this testset I've put in a `exit()` so the solverdummy doesn't run, because running the solverdummy would have to access the correct SolverInterface. Until the path works, I'll leave it like this. 
The one test that doesn't work is the [`readWriteScalarData` test](https://github.com/precice/PreCICE.jl/blob/599f99785a4c1f9c03ec22fc4cef8fa4312e853f/test/test_functioncalls.jl#L122-L129). Here a scalar value is written to [the `fake_read_write_buffer`](https://github.com/precice/PreCICE.jl/blob/599f99785a4c1f9c03ec22fc4cef8fa4312e853f/test/SolverInterface.c#L273). This works without problem as far as my debugging shows. 
However, [reading the data](https://github.com/precice/PreCICE.jl/blob/599f99785a4c1f9c03ec22fc4cef8fa4312e853f/test/test_functioncalls.jl#L126) doesn't. The Problem probably lies somewhere in handling references and pointers [with the C code](https://github.com/precice/PreCICE.jl/blob/599f99785a4c1f9c03ec22fc4cef8fa4312e853f/test/SolverInterface.c#L301-L304).
[`PreCICE.jl`](https://github.com/precice/PreCICE.jl/blob/599f99785a4c1f9c03ec22fc4cef8fa4312e853f/src/PreCICE.jl#L1244) passes a reference to a double into the C function which then is supposed to rewrite the data where the pointer is pointing to. 
In the [Python bindings](https://github.com/precice/python-bindings/blob/cd13eccb75847f0247dc8c6ecb9b8aeef86398ce/test/SolverInterface.cpp#L363), the pointer is different, but this is C++ code instead of C and I don't have enough experience with C/C++ to fix this error. 
The tests should also resemble how it is implemented in the original SolverInterface so the orientation shouldn't only be the python bindings.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ binarybuilder
 solverdummy/*.log
 solverdummy/*.json
 solverdummy/precice-run/
+*.o
+*.so
```
Comment (IshaanDesai) [inline .gitignore:7]: Do we really need to remove these? Does the build process no longer have these files?

File: README.md [hunk 2]
```diff
@@ -24,21 +24,26 @@ julia> using PreCICE
...
 
-```julia
+```julia-repl
 julia> ]
 pkg> add <path-to-repository>
 Then exit the package mode with 🔙 or Ctrl + c
 julia> using PreCICE
 ```
 
+You can alternatively also install a specific branch from this repository:
```
Comment (IshaanDesai) [inline README.md:42]:
```suggestion
Alternatively, you can install a specific branch of this repository with the following command:
```

File: README.md [hunk 3]
```diff
@@ -62,16 +67,20 @@ julia> using PreCICE
 
 The [solverdummy](https://github.com/precice/julia-bindings/tree/main/solverdummy) shows an example of how to use the Julia bindings for preCICE.
 
-## Testing new features that are on branches of this repository
-
-To use a certain branch of this package, add `#branchname` after the package url, for example if the branch `mpi-parallelization` is to be tested:
+## Testing PreCICE.jl
 
+To test the bindings run:
```
Comment (IshaanDesai) [inline README.md:72]:
```suggestion
To test the bindings, run:
```

File: test/Makefile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+OBJECTS=$(SOURCES:.c=.o)
+ 
+.c.o:
+	$(CC) $(CFLAGS) $< -o $@ 
+ 
+lib: $(OBJECTS)
+	$(CC) -shared -fPIC -o libprecice.so $(OBJECTS)
+ 
+clean:
+	rm *.o *.so
```
Comment (IshaanDesai) [inline test/Makefile:15]: Add newline at the end of the file if required

File: test/SolverInterface.c [hunk 5]
```diff
@@ -0,0 +1,321 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Im not sure if the functions with fake_read_write_buffer are correct
+// pointers also not sure in readscalardata,actionwriteinitialdata.
+// in readscalardata, the python bindings have `&value` instead of `*value` but this gives an error on my system
```
Comment (IshaanDesai) [inline test/SolverInterface.c:8]: Can we remove these comments?

**Review by IshaanDesai — APPROVED** [review]
Only a few minor comments, otherwise its good to go :+1: Good job here, we now have a proper testing infrastructure!

File: .gitignore (hunk 1 above)
Comment (erikscheurer) [reply .gitignore:7]: The testing process produces these files so they should be excluded from git

**Pull request merged**